oauthlib/oauth2/rfc6749/endpoints/base.py

```diff
@@ -62,6 +66,16 @@ def _raise_on_unsupported_token(self, request):
             request.token_type_hint not in self.supported_token_types):
             raise UnsupportedTokenTypeError(request=request)
 
+    def _raise_on_bad_post_request(self, request):
+        """Raise if invalid POST request received
+        """
+        if request.http_method.lower() == 'post':
+            query_params = CaseInsensitiveDict(dict(urldecode(request.uri_query)))
+            for param in self._blacklist_query_params:
```

Comment:
Maybe a last comment for me: we agreed that the specifications require to have all OAuth2 POST parameters in the body and not in the query.

So, instead of focusing only on the "sensitive" parameters like secret+verifier, what about if you replace `self._blacklist_query_params` by `request._params.keys()` ? 

That way, we are again closer to the specification. 
Thoughts?

Reply:
I just pushed with banning any and all query parameters; on the target endpoints only of course, not just OAuth related. Would that work?

Reply:
That's great!